src/rule_tester.rs

```diff
@@ -31,55 +37,55 @@ impl RuleTester {
     }
 
     fn run_valid_test(&self, valid_test: &RuleTestValid) {
-        let tmp_dir = TempDir::new("valid_test").unwrap();
-        let test_filename = "tmp.rs";
-        fs::write(tmp_dir.path().join(test_filename), &valid_test.code).unwrap();
-        Command::cargo_bin("tree-sitter-lint")
-            .unwrap()
-            .args(["--rule", &self.rule.name])
-            .current_dir(tmp_dir.path())
-            .assert()
-            .success()
-            .stdout(predicate::str::is_empty());
+        let violations = crate::run_for_slice(
+            valid_test.code.as_bytes(),
+            "tmp.rs",
+            ConfigBuilder::default()
+                .rule(&self.rule.meta.name)
+                .language(SupportedLanguage::Rust)
```

Comment:
I haven't tried to really think about "language" yet

But it seems like having `language` as a field on `Config` may be a conceptual mismatch

Because it seems like you should be able to lint against multiple languages in a single linter run if you want to

So then that implies probably that "language-awareness" is driven by rules (and/or config in configuration files)?

Eg maybe each rule is responsible for declaring which language(s) it is applicable to?

And then instead of one big "aggregated tree-sitter query" we do one of those per language (out of all the languages that your enabled rules are for)? And maybe run those in parallel?

Reply:
> Eg maybe each rule is responsible for declaring which language(s) it is applicable to?

Yeah that makes sense to me.